### PR TITLE
bison and flex dependancies in gobject-introspection

### PR DESCRIPTION
**Pull request by andrewjpage**
from https://github.com/Homebrew/homebrew/pull/41775

### CODE REVIEW

File: Library/Formula/gobject-introspection.rb [hunk 1]
```diff
@@ -19,6 +19,11 @@ class GobjectIntrospection < Formula
   depends_on "libffi"
   depends_on "cairo" => :build if build.with? "tests"
 
+  if OS.linux?
+    depends_on "bison"
+    depends_on "flex"
+  end
+
```
Comment (sjackman) [inline Library/Formula/gobject-introspection.rb]: Please change this to
```ruby
depends_on "bison" unless OS.mac?
depends_on "flex" unless OS.mac?
```
and delete the blank line.

**Comment by sjackman** [comment]
Please report the exact error message that you see without this fix, so that the issue can be searched by error message.

**Comment by andrewjpage** [comment]
The errors are:
```
checking for flex... no
checking for lex... no
configure: error: flex not found but required
```
and 
```
checking for bison... no
configure: error: bison not found but required
```

**Comment by sjackman** [comment]
Merged. Thanks for your contribution, Andrew.

Please squash multiple commits into a single commit. I changed the commit log message to
```
gobject-introspection: Fix depends for Linuxbrew

Depends on bison and flex.

Fix
configure: error: flex not found but required
configure: error: bison not found but required
```

**Comment by sjackman** [comment]
Please keep the first line to fewer than 50 characters. Please include the error message in the commit log message.

**Pull request closed without merging**